src/Git.hs

```diff
@@ -410,6 +414,18 @@ runGit userConfig repoDir operation =
 
     DeleteTag t cont -> cont <$ callGitInRepo ["tag", "-d", refSpec t]
 
+    CheckOrphanFixups sha branch cont -> do
+      result <- let branch' = refSpec branch
+                    sha' = refSpec sha 
+                in callGitInRepo ["log", Text.unpack $ format "{}..{}" [branch',sha']]
+      case result of
+        Left (code, message) -> do
+          logWarnN $ format "git log failed with code {}: {}" (show code, message)
+          pure $ cont Nothing
+        Right logResponse -> do 
+          logWarnN "there is one ore more fixup commits not belonging to any other commit"
+          pure $ cont $ Just $ "fixup!" `isInfixOf` logResponse 
```

Comment:
Since we were talking about this yesterday, I was thinking that maybe this check needs a bit more specific. Right now, it would also reject a commit like this

```
Prevent orphan fixup commits

Refuse a merge on any commit containing `fixup!`.
```

because it contains the literal string `fixup!`. But that `fixup!` is not in a position where it would mean anything, so it should be allowed.

Arguably an edge case, sooner or later, all edge cases will get triggered.

What you can do instead is pass `--format='%s'` (%s meaning "subject", the header line of the commit message) to print one commit header per line and then check that no line starts with `fixup!`.

Reply:
Good point, thanks!